chainer/functions/loss/huber_loss.py

```diff
@@ -37,15 +40,20 @@ def forward(self, inputs):
         else:
             return y,
 
-    def backward(self, inputs, gy):
-        xp = cuda.get_array_module(*inputs)
+    def backward(self, indexes, grad_outputs):
+        x0, x1 = self.get_retained_inputs()
+        diff = x0 - x1
+        gy, = grad_outputs
+
+        xp = cuda.get_array_module(self.diff)
         mask = xp.abs(self.diff) <= self.delta
 
-        gx = xp.where(mask, self.diff, self.delta * xp.sign(self.diff))
-        gy_ = gy[0]
+        gx = chainer.functions.where(
+            mask, diff, self.delta * xp.sign(self.diff))
```

Comment:
Do we need `self.diff` for backward compatibility?  If not, `self.diff` can be replaced by `diff.array`.

Reply:
It is there simply to avoid having to recompute it in the backward pass (although it simply a subtraction and could be recomputed quite easily). Do you have a good reason why we'd want to change this?

Reply:
Sorry.  I wasn't very familiar with new style functions.
I think `diff = x0 - x1` also computes the value, and then `diff.array` will cost little.  Is it true?

By the way, is there any way to save `x0 - x1` in `Variable` at `forward`?  Then it would not have to retain inputs `x0`, `x1`.  In other words, once `self.diff` becomes `Variable`, this line can be
```
        gx = chainer.functions.where(
            mask, self.diff, self.delta * xp.sign(self.diff))
```

Reply:
In terms of performance, I think the difference is pretty small but you are right, I'll fix it!

As for the latter, I also think that there are other use-cases where we'd like to retain intermediate Variables but that is unfortunately not possible as far as I know.

Reply:
Removed `self.diff`!